### ISSUE TITLE
Single-hyphen long-form parameter names don't work with IntelliJ's "Run anything" feature

### ISSUE THREAD

**Issue by lptr** [gradle/gradle#26240]
### Current Behavior

For long-form parameter names Gradle supports both `--switch` (double hyphens) as well as `-switch` (single hyphen) syntax. We actively rely on this when recommending the command to run to update verification-metadata.xml here: https://github.com/gradle/gradle/blob/28d4baf20eee2f5cb43a1007261d165283af915f/gradle/verification-metadata.xml#L31

When you run Gradle via IntelliJ's “Run anything” feature (i.e. by pressing Ctrl+Ctrl and typing `gradle ...`), long-form parameters with single-hyphens don’t get recognized as switches, and somehow Gradle ends up “running” them as tasks.

I’m not sure where this is broken: does IntelliJ’s Gradle plugin do some sort of custom parsing? Do we mess up on our side? If so, where? Tooling API? Launcher? But it seems we should fix it.

To add insult to injury the bug causes these single hyphened switches that are considered as tasks to be captured in build scans, even though scans should not capture command-line arguments, only (real) task names.

### Expected Behavior

Running Gradle via IntelliJ's "Run anything" feature should work even when using single-hyphen long-form parameters like `-build-cache` etc. The parameters should not be captured in build scans.

### Steps to Reproduce

You can reproduce the behavior by running the command-line in the verification XML above; it will fail with not finding a task called `-write-verification-metadata`.

### Gradle version

8.3

### Build scan URL (optional)

https://ge.gradle.org/s/lconyryl77zcm

**Comment by donat** [comment]
To make Gradle more idiomatic, I think think the long-term solution should be to get rid of the single hyphen option altogether.

**Comment by lptr** [comment]
Yeah, I like that approach. I think the current undocumented acceptance of single-hyphened long-form parameters is just confusing.
